src/Presenter/PsAccountsPresenter.php

```diff
@@ -71,10 +75,21 @@ public function present($psx)
      */
     public function getEmail()
     {
-        return null !== \Tools::getValue('adminToken')
-            && !empty(\Tools::getValue('adminToken'))
-            && null !== \Tools::getValue('email')
-            && !empty(\Tools::getValue('email')) ? \Tools::getValue('email') : '';
+        if (
+            null !== \Tools::getValue('email')
+            && !empty(\Tools::getValue('email'))
+        ) {
+            if (
+                null !== \Tools::getValue('adminToken')
+                && !empty(\Tools::getValue('adminToken'))
+            ) {
+                $this->getRefreshTokenWithAdminToken();
+            }
+
+            return \Tools::getValue('email');
+        } else {
+            return '';
+        }
```

Comment:
Try to always avoid to using if in if as much as possible in order to make it more readable and understandable.

```suggestion
        if (
            null !== \Tools::getValue('email')
            && !empty(\Tools::getValue('email'))
        ) {
            return false;
        }
        
        if (
            null !== \Tools::getValue('adminToken')
            && !empty(\Tools::getValue('adminToken'))
        ) {
            $this->getRefreshTokenWithAdminToken();
        }

        return \Tools::getValue('email');
```